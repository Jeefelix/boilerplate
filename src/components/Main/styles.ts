import styled from "styled-components";

export const Wrapper = styled.main`
  background-color: #ffebcd;
  color: #1c1c1c;
  width: 100%;
  height: 100%;
  text-align: center;
  align-items: center;
  justify-content: center;
  display: flex;
  flex-direction: column;
  padding: 3rem;
`;
export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`;

export const Title = styled.h1`
  font-size: 2.5rem;
`;

export const Description = styled.h2`
  font-size: 2rem;
  font-wiight: 400;
`;

export const Illustration = styled.img`
  width: min(30rem, 100%);
  margin-top: 2.5rem;
`;
