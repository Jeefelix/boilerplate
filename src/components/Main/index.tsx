import * as S from "./styles";

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/3Logos.svg"
      alt="circulo preenchido da cor preta com a letra N sem serifa vasada, depois um emoji de unha pintada, dentro de um sinal de maior e menor, por ultimo, o simbolo de atomo em azul, com fundo preto, dentro de uma forma redonda"
    />
    <S.Title>Boilerplate</S.Title>
    <S.Description>
      Typescript, React.js, Next.js and Styled Components
    </S.Description>
    <S.Illustration
      src="/img/pusheen.svg"
      alt="desenho de gatinho cinza com vergonha, com as patinhas em direção as bochechas"
    />
  </S.Wrapper>
);
export default Main;
