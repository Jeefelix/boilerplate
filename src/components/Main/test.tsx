/* eslint-disable prettier/prettier */
import { render, screen } from "@testing-library/react";

import Main from "../Main";

describe("<Main>", () => {
  it("should render the heading", () => {
      const {container}= render(<Main/>)

      expect(
          screen.getByRole('heading',{name:/Boilerplate/i})
      ).toBeInTheDocument()

      expect(container.firstChild).toMatchSnapshot()
  });

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': 'rgb(255, 235, 205)' })
  });

});
