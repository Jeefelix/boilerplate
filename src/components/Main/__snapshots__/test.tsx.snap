// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main> should render the heading 1`] = `
.c0 {
  background-color: #ffebcd;
  color: #1c1c1c;
  width: 100%;
  height: 100%;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 3rem;
}

.c1 {
  width: min(25rem,100%);
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-wiight: 400;
}

.c4 {
  width: min(30rem,100%);
  margin-top: 2.5rem;
}

<main
  class="c0"
>
  <img
    alt="circulo preenchido da cor preta com a letra N sem serifa vasada"
    class="c1"
    src="/img/3Logos.png"
  />
  <h1
    class="c2"
  >
    Boilerplate
  </h1>
  <h2
    class="c3"
  >
    Typescript, React.js, Next.js and Styled Components
  </h2>
  <img
    alt="desenho de gatinho cinza feliz sorrindo e acenando, com chapéu de aniversário"
    class="c4"
    src="/img/gatinhoAniversário.gif"
  />
</main>
`;
